chia/cmds/start_funcs.py

```diff
@@ -30,13 +35,21 @@ async def create_start_daemon_connection(root_path: Path) -> Optional[DaemonProx
         await asyncio.sleep(1)
         # it prints "daemon: listening"
         connection = await connect_to_daemon_and_validate(root_path)
+        if connection and passphrase:
```

Comment:
is unlocking not needed in the case where connection is not None ?

Reply:
I think I understand your question in regard to the first check on line 25. That's a good catch. It would be an uncommon case to hit that scenario, but I can easily restructure the code to handle both scenarios (daemon already launched and waiting for unlock, or daemon not launched and will need unlocking).

In practice, the only way I think this scenario could play out is if the GUI launched the daemon, presented the passphrase dialog (still a WIP), and while leaving the dialog open, the user started a service on the command line.

Fix incoming.